### DISCUSSION TITLE
Disable built-in package management for shared environments

### DISCUSSION THREAD

**Discussion by jheinen** [fonsp/Pluto.jl#2204] [Pluto development]
Is there a way to disable Pluto's package management **outside** the notebook file, e.g. when Julia is started?

In a shared Julia environment (on a "Pluto hub") it sometimes makes more sense if you could switch off the internal package management, e.g. by using an environment variable, even if this does not necessarily guarantee the reproducibility of the notebooks.

**Comment by disberd** [comment]
Unfortunately there is no flag or way to do this outside the notebook for the time beeing.
I am curious though what is you specific use case and why are you not satisfied with just having a cell with `Pkg.activate` to activate the desired environment.

I think the PkgManager is great not only for reproducibility but also aids in trying out packages without the need of messing with the underlying envrionment yourself.
In the few cases where the manual project management is preferred, the single `Pkg.activate` cell is quite un-obstrusive, it just stays in the notebook so you don't have to re-type it everytime you open it and is a clear reminder that the Pkg is deactivated.

**Reply by jheinen** [reply]
Thanks for the answer.

I did not mean to criticize package management in any way - I am well aware of the merits. In any case, in our Pluto hub - which may be a special case - a central option to disable package management would be beneficial for then **end users** - that's where the criticism came from.

Although everything has become much better, package load and precompile times (in Julia in general) are still criticized by end users.

**Comment by disberd** [comment]
Hi @jheinen,
I never intended your question as criticism, I just wanted to understand better your use case to see if there was an alternative solution to disabling the PkgManager.
I am not familiar with your Pluto hub, is it a central Pluto server where multiple people can open notebooks?
For what concerns the package load and precompile times, I don't think the PkgManager is by itself contributing to this. 
It does add and precompile packages if you use it (and only upon change of packages and not every time it a notebook is reloaded), but again putting a reminder to eventualy run a cell with Pkg.activate is more flexible.

Is the pluto instance ran just once in your PlutoHub or users start the pluto server themselves?

You can hackishly disable the pkgmanager by running `@eval Pluto use_plutopkg(t::NotebookTopology) = false` after importing Pluto and before doing `Pluto.run()`.
It's probably not the best solution (as I mentioned above I still think `Pkg.activate()` to be significantly better without much burden on the user) but if you really need a server-wide disable of the pkgmanager it's probably much faster than opening a PR and hoping/waiting it gets accepted
